diff --git a/epsacrop.module b/epsacrop.module
index 544cc68..01f5dce 100644
--- a/epsacrop.module
+++ b/epsacrop.module
@@ -667,14 +667,21 @@ function epsacrop_element_process($element, $form_state, $form) {
   }
 
   if ($element['#type'] == 'media') {
-    //for media, need to include the media elements even if the file is not defined, because it's not defined when file initially selected
-    $element['#attached']['js'][] = array('type' => 'setting', 'data' => array( 'epsacrop_dialog' => array($element['#id'] => $setting)));
+    // For media, we need to include the media elements even if the file is not
+    // defined, because it's not defined when the file is initially selected.
+    $element_id = drupal_clean_css_identifier($element['#name']);
+    $element['#attached']['js'][] = array('type' => 'setting', 'data' => array( 'epsacrop_dialog' => array($element_id => $setting)));
     $element['#attached']['js'][] = drupal_get_path('module', 'epsacrop') . '/js/epsacrop-media.js';
     $element['epsacrop'] = array(
       '#type' => 'link',
       '#href' => '',
       '#title' => t('Manage image crops'),
-      '#attributes' => array('class' => array('button', 'manage-crop')),
+      '#attributes' => array(
+        'class' => array('button', 'manage-crop'),
+        // We need to include the parent id because the Media widget doesn't
+        // include it itself.
+        'data-epsaid' => array($element_id),
+      ),
       '#options' => array('fragment' => FALSE, 'external' => TRUE),
       '#weight' => 0,
     );
diff --git a/js/epsacrop-media.js b/js/epsacrop-media.js
index 75a4c56..093330e 100644
--- a/js/epsacrop-media.js
+++ b/js/epsacrop-media.js
@@ -23,12 +23,7 @@ Drupal.behaviors.EPSACropMediaElement = {
       // When someone clicks the link to manage EPSA crops
       epsaButton.bind('click', function (e) {
         e.preventDefault();
-        var elem = $(this).parents('.media-widget');
-        // Handle media widgets with an unlimited cardinality.
-        if ($(this).parent('td').length === 1) {
-          elem = $(this).parents('tr').find('.media-widget');
-        }
-        var fileInfo = epsaDialogSettings[elem.attr('id')];
+        var fileInfo = epsaDialogSettings[$(this).attr('data-epsaid')];
         var fid = fidField.val();
         if(!fileInfo.fid || fileInfo.fid != fid) {
           //if no file info or file has been replaced, get if via ajax using the fid
